src/management-system-v2/lib/data/roles.ts

```diff
@@ -72,3 +73,13 @@ export async function getRoles(environmentId: string) {
     return userError("Something wen't wrong");
   }
 }
+
+export async function getUserRoles(environmentId: string, userId: string) {
+  try {
+    return _getUserRoles(userId, environmentId);
```

Comment:
If you don't pass an ability to the db function the persmissions won't be checked

```suggestion
    const { ability } = await getCurrentEnvironment(environmentId);

    return _getUserRoles(userId, environmentId, ability);

```

Reply:
I added the check